radicle-node/src/worker/fetch.rs

```diff
@@ -370,17 +398,34 @@ impl<'a> StagingPhaseFinal<'a> {
         let head = production.set_identity_head()?;
         log::debug!(target: "worker", "'refs/rad/id' for {} set to {head}", production.id);
 
+        // N.b. This is to prevent us from shooting ourselves in the
+        // foot with storage inconsistencies.
+        debug_assert_matches!(
+            production.validate(),
+            Ok(()),
+            "repository {} is not valid",
+            production.id,
+        );
+
```

Comment:
```suggestion
        debug_assert_matches!(
            production.validate(),
            Ok(()),
            "repository {} is not valid",
            production.id,
        );

```

I did not find an alternative to this, but here are a possible solution.

```diff
+++ b/radicle-node/src/worker/fetch.rs
@@ -13,7 +13,7 @@ use radicle::storage::git::Repository;
 use radicle::storage::refs::IDENTITY_BRANCH;
 use radicle::storage::{Namespaces, RefUpdate, Remote, RemoteId};
 use radicle::storage::{ReadRepository, ReadStorage, WriteRepository, WriteStorage};
-use radicle::{debug_assert_matches, git, Storage};
+use radicle::{git, Storage};
 
 /// The initial phase of staging a fetch from a remote.
 ///
@@ -400,11 +400,12 @@ impl<'a> StagingPhaseFinal<'a> {
 
         // N.b. This is to prevent us from shooting ourselves in the
         // foot with storage inconsistencies.
-        debug_assert_matches!(
-            production.validate(),
-            Ok(()),
-            "repository {} is not valid",
+        let validation = production.validate();
+        debug_assert!(
+            validation.is_ok(),
+            "repository {} is not valid: {:?}",
             production.id,
+            validation,
         );
 
         Ok((updates, remotes))

```